app/controllers/application_controller.rb

```diff
@@ -9,6 +9,6 @@ def ensure_domain
     return if request.host == ENV['PREFERRED_DOMAIN']
 
     Rails.logger.info("Handling Domain Redirect: #{request.host}")
-    redirect_to "https://#{ENV['PREFERRED_DOMAIN']}", status: :moved_permanently
+    redirect_to "https://#{ENV['PREFERRED_DOMAIN']}", status: :moved_permanently, allow_other_host: true
```

Comment:
I'm comfortable with this since we control the redirect target as you noted. If we redirected to a URL supplied target it could be used to trick people to think they were on our domain much like ezproxy was targeted before it moved to a safelist model for redirects. Makes sense this now requires allowing explicitly.